src/dust/dust.cpp

```diff
@@ -320,6 +339,379 @@ TaskStatus FluxSource(MeshData<Real> *md, const Real dt) {
+                        int &lmax2) {
+            geometry::Coords<GEOM> coords(vmesh.GetCoordinates(b), k, j, i);
+            const Real vol00 = coords.Volume();
+            for (int n = 0; n < nspecies; ++n) {
+              Real &dens_d = vmesh(b, dust::cons::density(n), k, j, i);
+              lsum += dens_d * vol00;
+            }
+            lmax1 = std::max(lmax1, nCalls(k, j, i));
+            for (int n = nspecies - 1; n >= 0; --n) {
+              Real &dens_d = vmesh(b, dust::cons::density(n), k, j, i);
+              if (dens_d > dfloor) {
+                lmax2 = std::max(lmax2, n);
+                break;
+              }
+            }
+          },
+          Kokkos::Sum<Real>(sumd0), Kokkos::Max<int>(maxCalls1),
+          Kokkos::Max<int>(maxSize1));
+      massd += sumd0;
+      maxCalls = std::max(maxCalls, maxCalls1);
+      maxSize = std::max(maxSize, maxSize1);
+    } // end if (info_out_flag)
+  }   // loop of blocks
+
+  if (info_out_flag) {
+#ifdef MPI_PARALLEL
+    // over all processors
+    MPI_Reduce(MPI_IN_PLACE, &maxCalls, 1, MPI_INT, MPI_MAX, 0, MPI_COMM_WORLD);
+    MPI_Reduce(MPI_IN_PLACE, &maxSize, 1, MPI_INT, MPI_MAX, 0, MPI_COMM_WORLD);
+    MPI_Reduce(MPI_IN_PLACE, &massd, 1, MPI_PARTHENON_REAL, MPI_SUM, 0, MPI_COMM_WORLD);
+#endif // MPI_PARALLEL
+
+    if (parthenon::Globals::my_rank == 0) {
+      std::string fname;
+      // fname.assign(pin->GetString("parthenon/job", "problem_id"));
+      fname = "artemis";
+      fname.append("_coag_info.dat");
+      static FILE *pfile = NULL;
+
+      // The file exists -- reopen the file in append mode
+      if (pfile == NULL) {
+        if ((pfile = std::fopen(fname.c_str(), "r")) != nullptr) {
+          if ((pfile = std::freopen(fname.c_str(), "a", pfile)) == nullptr) {
+            PARTHENON_FAIL("Error output file could not be opened");
+          }
+          // The file does not exist -- open the file in write mode and add headers
+        } else {
+          if ((pfile = std::fopen(fname.c_str(), "w")) == nullptr) {
+            PARTHENON_FAIL("Error output file could not be opened");
+          }
+          std::string label = "# time dt maxCall end_maxSize beg_massSize ";
+          label.append("end_total_massd beg_total_massd diff_massd \n");
+          std::fprintf(pfile, label.c_str());
+        }
+      }
+      std::fprintf(pfile, "  %e ", time);
+      std::fprintf(pfile, "  %e ", dt);
+      std::fprintf(pfile, "  %d  %d  %d ", maxCalls, maxSize, maxSize0);
+      std::fprintf(pfile, "  %e  %e  %e", massd, massd0, (massd - massd0) / massd);
+      std::fprintf(pfile, "\n");
+    }
+  } // end if (info_out_flag)
+
+  return TaskStatus::complete;
+}
+
+//----------------------------------------------------------------------------------------
+//! \fn  TaskCollection Dust::OperatorSplitDust
+//! \brief dust operator split task collection
+template <Coordinates GEOM>
+TaskCollection OperatorSplitDustSelect(Mesh *pm, parthenon::SimTime &tm) {
```

Comment:
This is the function that I'm suggesting be put into the driver class. You then have access to `SimTime` and other things.